_pytest/nodes.py

```diff
@@ -187,18 +187,34 @@ def find_markers(self, name):
         """find all marks with the given name on the node and its parents
 
         :param str name: name of the marker
-        :returns: iterator over marks matching the name
+        :returns: iterator over marks matching the name"""
+        return map(itemgetter(1), self.find_markers_with_node(name))
+
+    def find_markers_with_node(self, name):
+        """find all marks with the given name on the node and its parents
+
+        :param str name: name of the marker
+        :returns: iterator over (node, mark) matching the name
         """
         for node in reversed(self.listchain()):
             for mark in node._markers.find(name):
-                yield mark
+                yield node, mark
 
     def iter_markers(self):
         """
         iterate over all markers of the node
         """
         return chain.from_iterable(x._markers for x in reversed(self.listchain()))
 
+    def iter_markers_with_node(self):
```

Comment:
Same comment about, seems confusing. Again it seems having a single version would be less confusing (returning an iterator of `(node, mark)`).

We might even consider to drop the "find by name" versions and let users do the search themselves; this is very easy to do with a list/generator comprehension. We can always add more API later, but removing them is always problematic as we know.

Reply:
i would like to introduce those apis as experimental and leave them as such for at least 6 months

Reply:
I can also be found for having fewer variations of this available.  A minimalistic API would just be `iter_markers` which yields the `(node, mark)` tuples, is that a nice starting point?  It's pretty easy for the user to write the filter.  Especially since the implementations provided here don't provide anything major, it's not like `find_markers` is much more efficient then `iter_markers` with your own filter.

Reply:
ok, will do